crates/ruff_linter/src/rules/pycodestyle/rules/compound_statements.rs

```diff
@@ -225,14 +216,14 @@ pub(crate) fn compound_statements(
                     || while_.is_some()
                     || with.is_some()
                 {
-                    colon = Some((range.start(), range.end()));
+                    colon = Some((token.start(), token.end()));
```

Comment:
Nit: Maybe consider storing the `range` in `colon` (and the other fields) instead of a `start` `end` tuple.